edb/edgeql-parser/edgeql-parser-python/src/normalize.rs

```diff
@@ -188,7 +187,11 @@ pub fn normalize(text: &str) -> Result<Entry, Error> {
     }
 
     all_variables.push(variables);
-    let processed_source = serialize_tokens(&rewritten_tokens[..]);
+    let processed_source = if !all_variables.is_empty() {
```

Comment:
This is wrong, as `all_variables` is a `Vec` of `Vec` and so will not be empty.  Use the `counter <= var_idx` check below instead.

Reply:
Fixed